packages/react-components/react-persona/src/components/Persona/usePersona.ts

```diff
@@ -12,17 +14,95 @@ import type { PersonaProps, PersonaState } from './Persona.types';
  * @param ref - reference to root HTMLElement of Persona
  */
 export const usePersona_unstable = (props: PersonaProps, ref: React.Ref<HTMLElement>): PersonaState => {
-  return {
-    // TODO add appropriate props/defaults
+  const {
+    className,
+    presenceOnly = false,
+    sizing = 'fixed',
+    style,
+    textPosition = 'after',
+    ...avatarSlotProps
+  } = props;
+  const numTextLines =
+    (props.primaryText ? 1 : 0) +
+    (props.secondaryText ? 1 : 0) +
+    (props.tertiaryText ? 1 : 0) +
+    (props.quaternaryText ? 1 : 0);
+
+  let avatarSize = avatarSlotProps.size;
+  if (sizing === 'scaled') {
+    if (numTextLines === 1) {
+      avatarSize = 16;
+    } else if (numTextLines === 2) {
+      avatarSize = 32;
+    } else if (numTextLines === 3) {
+      avatarSize = 56;
+    } else {
+      avatarSize = 72;
+    }
+  }
+
+  const state: PersonaState = {
+    numTextLines,
+    presenceOnly,
+    sizing,
+    textPosition,
+
     components: {
-      // TODO add each slot's element type or component
       root: 'div',
+      avatar: Avatar,
+      presence: PresenceBadge,
+      primaryText: 'span',
+      secondaryText: 'span',
+      tertiaryText: 'span',
+      quaternaryText: 'span',
     },
-    // TODO add appropriate slots, for example:
-    // mySlot: resolveShorthand(props.mySlot),
-    root: getNativeElementProps('div', {
-      ref,
-      ...props,
+
+    root: resolveShorthand(props.root, {
+      required: true,
+      defaultProps: {
+        style,
+        className,
+      },
+    }),
+    avatar: resolveShorthand(props.avatar, {
+      required: true,
+      defaultProps: {
+        ref,
+        size: avatarSize,
+        ...avatarSlotProps,
+      },
+    }),
+    presence: resolveShorthand(props.presence, {
+      required: false,
+    }),
+    primaryText: resolveShorthand(props.primaryText, {
+      required: false,
+    }),
+    secondaryText: resolveShorthand(props.secondaryText, {
+      required: false,
+    }),
+    tertiaryText: resolveShorthand(props.tertiaryText, {
+      required: false,
+    }),
+    quaternaryText: resolveShorthand(props.quaternaryText, {
+      required: false,
     }),
   };
+
+  if (!presenceOnly && state.avatar && state.presence) {
+    state.avatar.badge = state.presence;
+    state.presence = undefined;
+  } else if (state.presence) {
+    if (sizing === 'scaled') {
+      if (numTextLines === 1) {
+        state.presence.size = 'small';
+      } else if (numTextLines === 2) {
+        state.presence.size = 'large';
+      } else {
+        state.presence.size = 'extra-large';
+      }
+    }
+  }
```

Comment:
These values shouldn't override user-specified values (like state.presence.size) if they were provided. One way to do it is to calculate the values ahead of time and pass them in `defaultProps` in `resolveShorthand`. Or at least check that the prop isn't set before overwriting it.